### ISSUE TITLE
Improve RoutePrefix documentation

### ISSUE THREAD

**Issue by Injac** [Azure/azure-mobile-services#396]
If you set a custom route-prefix like:

 [RoutePrefix("Mothership")]

on a custom controller and you try to call an action, that is attributed as well like this:

[Route("connectioncheck")]

The usual way would be:

var result = await App.client.InvokeApiAsync<bool>("Mothership/connectioncheck",HttpMethod.Get,null);

Which will not work, until you change the controller-wide attribute to:

 [RoutePrefix("api/Mothership")]

Maybe this should be documented somewhere. Client adds the prefix "/api/..." to every call. Therefore the call will fail, if you add a custom attribute, because the route lacks the "api" part.

**Comment by paulbatum** [comment]
The SEO for this issue is pretty good so someone running into the issue should find the answer here and be unblocked. Closing.

**Comment by Singh400** [comment]
I agree with @Injac this should be documented a bit more explicitly.

I had to dig through the code for `MobileServiceClient` to understand what was going on.

Like @Injac, I had a routing problem and found out `InvokeApiAsync` prepends `api/`, this is implicit behaviour and I feel should be documented explicitly.

I also found out that I can override this implicit behaviour by making sure my usage starts with `/` - again this is not documented anywhere. I had to read the [source code](https://github.com/Azure/azure-mobile-services/blob/4c3556d3fd3c89cacf9645b936ed495ec882eb02/sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/MobileServiceClient.cs#L589) to find out I could do that.

Cheers,
Indy
